pkg/gadgets/trace/tcpretrans/types/types.go

```diff
@@ -34,89 +32,38 @@ type Event struct {
-
-	/* Destination IP resolved by kubeipresolver  */
-	DstKind      eventtypes.RemoteKind `json:"dstKind,omitempty" column:"dstKind,maxWidth:5,hide,order:3100"`
-	DstNamespace string                `json:"dstNamespace,omitempty" column:"dstns,hide,order:3101"`
-	DstName      string                `json:"dstName,omitempty" column:"dstname,hide,order:3102"`
-}
-
-func (e *Event) SetLocalPodDetails(owner, hostIP, podIP string, labels map[string]string) {
-	// Unused
+	SrcEndpoint eventtypes.Endpoint `json:"src,omitempty" column:"src"`
+	DstEndpoint eventtypes.Endpoint `json:"dst,omitempty" column:"dst"`
 }
 
-func (e *Event) GetRemoteIPs() []string {
-	return []string{e.Saddr, e.Daddr}
-}
-
-func (e *Event) SetEndpointsDetails(endpoints []eventtypes.EndpointDetails) {
-	if len(endpoints) != 2 {
-		return
-	}
-	e.SrcName = endpoints[0].Name
-	e.SrcNamespace = endpoints[0].Namespace
-	e.SrcKind = endpoints[0].Kind
-
-	e.DstName = endpoints[1].Name
-	e.DstNamespace = endpoints[1].Namespace
-	e.DstKind = endpoints[1].Kind
+func (e *Event) GetEndpoints() []*eventtypes.Endpoint {
+	return []*eventtypes.Endpoint{&e.SrcEndpoint, &e.DstEndpoint}
 }
 
 func GetColumns() *columns.Columns[Event] {
 	cols := columns.MustCreateColumns[Event]()
 
-	// Virtual column for the source and destination endpoints
-	err := cols.AddColumn(columns.Attributes{
-		Name:    "src",
-		Visible: true,
-		Width:   30,
-		Order:   2000,
-	}, func(e *Event) string {
-		switch e.SrcKind {
-		case eventtypes.RemoteKindPod:
-			return "p/" + e.SrcNamespace + "/" + e.SrcName + ":" + fmt.Sprint(e.Sport)
-		case eventtypes.RemoteKindService:
-			return "s/" + e.SrcNamespace + "/" + e.SrcName + ":" + fmt.Sprint(e.Sport)
-		case eventtypes.RemoteKindOther:
-			return "o/" + e.Saddr + ":" + fmt.Sprint(e.Sport)
-		}
-		return e.Saddr + ":" + fmt.Sprint(e.Sport)
-	})
-	if err != nil {
-		panic(err)
-	}
-	err = cols.AddColumn(columns.Attributes{
-		Name:    "dst",
-		Visible: true,
-		Width:   30,
-		Order:   3000,
-	}, func(e *Event) string {
-		switch e.DstKind {
-		case eventtypes.RemoteKindPod:
-			return "p/" + e.DstNamespace + "/" + e.DstName + ":" + fmt.Sprint(e.Dport)
-		case eventtypes.RemoteKindService:
-			return "s/" + e.DstNamespace + "/" + e.DstName + ":" + fmt.Sprint(e.Dport)
-		case eventtypes.RemoteKindOther:
-			return "o/" + e.Daddr + ":" + fmt.Sprint(e.Dport)
-		}
-		return e.Daddr + ":" + fmt.Sprint(e.Dport)
-	})
-	if err != nil {
-		panic(err)
-	}
+	eventtypes.MustAddVirtualEndpointColumn(
+		cols,
+		columns.Attributes{
+			Name:    "src",
+			Visible: true,
+			Width:   30,
+			Order:   2000,
+		},
+		func(e *Event) eventtypes.Endpoint { return e.SrcEndpoint })
+	eventtypes.MustAddVirtualEndpointColumn(
+		cols,
+		columns.Attributes{
+			Name:    "dst",
+			Visible: true,
+			Width:   30,
+			Order:   3000,
+		},
+		func(e *Event) eventtypes.Endpoint { return e.DstEndpoint })
```

Comment:
I think Jose made a similar comment before, not sure. I strongly think the last `)` should go in a new line to make this more readable.

```go
	eventtypes.MustAddVirtualEndpointColumn(
		tcpColumns,
		columns.Attributes{
			Name:    "src",
			Visible: true,
			Width:   30,
			Order:   2000,
		},
		func(e *Event) eventtypes.Endpoint { return e.SrcEndpoint },
	)
```